src/core/border_router/routing_manager.cpp

```diff
@@ -2237,16 +2244,52 @@ void RoutingManager::OnMeshPrefixArray::MarkAsDeleted(const OnMeshPrefix &aPrefi
 
 RoutingManager::Nat64PrefixManager::Nat64PrefixManager(Instance &aInstance)
     : InstanceLocator(aInstance)
+    , mEnabled(false)
     , mTimer(aInstance, HandleTimer)
 {
     mInfraIfPrefix.Clear();
     mLocalPrefix.Clear();
     mPublishedPrefix.Clear();
 }
 
+void RoutingManager::Nat64PrefixManager::SetEnabled(bool aEnabled)
+{
+    mEnabled = aEnabled;
+
+    if (aEnabled)
+    {
+        if (Get<RoutingManager>().IsRunning())
+        {
+            Start();
+        }
+    }
+    else
+    {
+        Stop();
+    }
+}
+
+void RoutingManager::Nat64PrefixManager::OnRoutingManagerStart(void)
+{
+    if (IsEnabled())
+    {
+        Start();
+    }
+}
+
+void RoutingManager::Nat64PrefixManager::OnRoutingManagerStop(void)
+{
+    Stop();
+}
+
```

Comment:
I think we can combine this and use `Start()` and `Stop()`.

```c++
void RoutingManager::Nat64PrefixManager::Start(void)
{
    VerifyOrExit(mEnabled);
    mTimer.Start(0);
exit:
     return;
}
```
We can assume `RoutingManager` would invoke `Start()` and `Stop()` in order (i.e. not multiple calls to `Start()` without a `Stop()`.).